OutOfSchool/OutOfSchool.BusinessLogic/Services/ThumbnailProcessor/ThumbnailProcessingService.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Microsoft.Extensions.Options;
+using OutOfSchool.BusinessLogic.Models;
+using OutOfSchool.ExternalFileStore;
+using OutOfSchool.ExternalFileStore.Models;
+using SkiaSharp;
+
+namespace OutOfSchool.BusinessLogic.Services.ThumbnailProcessor;
+public class ThumbnailProcessingService : IThumbnailProcessingService
+{
+    private readonly IImageService imageService;
+    private readonly IImageStorage imageStorage;
+    private readonly ILogger<ThumbnailProcessingService> logger;
+    private readonly ThumbnailGenerationOptions options;
+    public ThumbnailProcessingService(
+        IImageService service,
+        IImageStorage imageStorage,
+        ILogger<ThumbnailProcessingService> logger,
+        IOptions<ThumbnailGenerationOptions> options)
+    {
+        this.imageService = service;
+        this.imageStorage = imageStorage;
+        this.logger = logger;
+        this.options = options.Value;
+    }
+    public async Task<bool> HasThumbnail(string imageId)
+    {
+        var thumbnailId = GetThumbnailId(imageId);
+        var thumbnail = await imageService.GetByIdAsync(thumbnailId);
```

Comment:
`Exists` method that gets object to check if it exists is expensive if we're talking bout file operations in a distributed system.

You can add an `exists` method to image storage / image service that uses the StatObject for Minio and Similar operation for GCP (so we have equal implementation)